bootstrap/kubeadm/internal/controllers/token.go

```diff
@@ -101,6 +102,9 @@ func refreshToken(ctx context.Context, c client.Client, token string, ttl time.D
 func shouldRotate(ctx context.Context, c client.Client, token string, ttl time.Duration) (bool, error) {
 	secret, err := getToken(ctx, c, token)
 	if err != nil {
+		if apierrors.IsNotFound(err) {
```

Comment:
could you please add a code comment that explains why this is needed (and that we normally don't expect to ever reach this)

Reply:
done